### PR TITLE
Make backend device checking more robust

### PR DESCRIPTION
**Pull request by theasianpianist**
**- How I did it**
* Treat devices that are ToRRouters (ToRRouters and BackEndToRRouters) the same when rendering templates
    - Except for BackEndToRRouters belonging to a storage cluster, since these devices have extra sub-interfaces created
* Treat devices that are LeafRouters (LeafRouters and BackEndLeafRouters) the same when rendering templates
**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by theasianpianist** [comment]
retest vsimage please

**Comment by theasianpianist** [comment]
retest vsimage please

**Review by prsunny — DISMISSED** [review]
lgtm.. Wait for @lguohan 's approval

**Comment by theasianpianist** [comment]
retest vsimage please

**Comment by theasianpianist** [comment]
retest vsimage please

**Comment by theasianpianist** [comment]
retest vsimage please

File: dockers/docker-router-advertiser/docker-router-advertiser.supervisord.conf.j2
Comment (lguohan) [inline dockers/docker-router-advertiser/docker-router-advertiser.supervisord.conf.j2:33]: this seems to me some behaviorial change if the device is MgmtTorRouter. I am not saying this is not a good change, but maybe you want to confirm with @sumukhatv on M0 device.

File: files/build_templates/buffers_config.j2
Comment (lguohan) [inline files/build_templates/buffers_config.j2:14]: can you check if this will have impact on MgmtTorRouter. it should not have any buffer configuration, I mean should not have any impact. But I am not 100% sure.

**Review by prsunny — APPROVED** [review]

**Review by sumukhatv — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Pull request merged**
